prog/vnet/cert_nexus.rb

```diff
@@ -0,0 +1,121 @@
+# frozen_string_literal: true
+
+require "acme-client"
+require "openssl"
+
+class Prog::Vnet::CertNexus < Prog::Base
+  subject_is :cert
+  semaphore :destroy
+
+  def self.assemble(hostname, dns_zone_id)
+    unless DnsZone[dns_zone_id]
+      fail "Given DNS zone doesn't exist with the id #{dns_zone_id}"
+    end
+
+    DB.transaction do
+      cert = Cert.create_with_id(hostname: hostname, dns_zone_id: dns_zone_id)
+
+      Strand.create(prog: "Vnet::CertNexus", label: "start") { _1.id = cert.id }
+    end
+  end
+
+  def before_run
+    when_destroy_set? do
+      hop_destroy unless %w[destroy].include?(strand.label)
+    end
+  end
+
+  label def start
+    register_deadline(:wait, 10 * 60)
+
+    private_key = OpenSSL::PKey::RSA.new(4096)
+    client = Acme::Client.new(private_key: private_key, directory: Config.acme_directory)
+    account = client.new_account(contact: "mailto:#{Config.acme_email}", terms_of_service_agreed: true, external_account_binding: {kid: Config.acme_eab_kid, hmac_key: Config.acme_eab_hmac_key})
+    order = client.new_order(identifiers: [cert.hostname])
+    authorization = order.authorizations.first
+    cert.update(kid: account.kid, private_key: private_key.to_s, order_url: order.url)
```

Comment:
I suggest using the `der` binary format. Unlike, say, attr_vault (which I think is better in at least this one way), it will get base64 encoded anyway by Sequel column encryption. No need to bloat the representation by double-stacking things. You might notice the acme-client transmits everything in der as well (at least the revocation code did)